src/omnisharp/server.ts

```diff
@@ -286,11 +342,24 @@ export class OmniSharpServer {
         });
     }
 
-    public stop(): Promise<void> {
-
-        while (this._disposables.length) {
-            this._disposables.pop().dispose();
+    // we could move this to a project updated observer and pass the server there. But we will need the event stream also there :(
+    private updateTracker = () => {
+        if (this.firstUpdateProject) {
```

Comment:
I don't really understand what this is trying to? The first time, it just sends the message, but subsequent project updates are debounced?

Reply:
The first time when we get one of the three events - 'onProjectAdded` , `onProjectChanged` and `onProjectRemoved` , the updateProjectInfo function is invoked directly. After this if we get any of the events, then we invoke the `updateProjectInfo` function via the debouncer. This is the functional equivalent of lodash.debounce with leading: true - https://css-tricks.com/debouncing-throttling-explained-examples/#article-header-id-1

Reply:
What's the consequence if you always debounce? The first message is delayed by debounce interval?

Reply:
If we keep getting the request then the status bar will not be displayed until there is a silence of 1500 ms. This way with leading true, we service the first request that we get so status bar shows something, and then when for the next requests we see a silence of 1500ms then we service them. Basically it is just so that we have something to display for the first time.

Reply:
Can you remind me what actual status/message this is displaying? If we can make this code  simpler and the consequence is that a message gets delayed by 1.5 seconds, I would be in favor of that. Up to you though...